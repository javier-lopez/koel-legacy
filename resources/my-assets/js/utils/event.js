import event from '@phanan/vuebus'
import { events as eventNames } from '@/config'

event.$names = eventNames

export { event }
