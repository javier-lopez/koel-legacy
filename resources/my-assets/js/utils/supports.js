import isMobile from 'ismobilejs'

/**
 * Check if AudioContext is supported by the current browser.
 *
 * @return {Boolean}
 */
export const isAudioContextSupported = (() => {
  // Apple device just doesn't love AudioContext that much.
  if (isMobile.apple.device) {
    return false
  }

  const AudioContext = (window.AudioContext ||
    window.webkitAudioContext ||
    window.mozAudioContext ||
    window.oAudioContext ||
    window.msAudioContext)

  if (!AudioContext) {
    return false
  }

  // Safari (MacOS & iOS alike) has webkitAudioContext, but is buggy.
  // @link http://caniuse.com/#search=audiocontext
  if (!(new AudioContext()).createMediaElementSource) {
    return false
  }

  return true
})()

/**
 * Checks if HTML5 clipboard can be used.
 * @return {Boolean}
 */
export const isClipboardSupported = 'execCommand' in document

/**
 * Checks if Media Session API is supported.
 * @return {Boolean}
 */
export const isMediaSessionSupported = 'mediaSession' in navigator
