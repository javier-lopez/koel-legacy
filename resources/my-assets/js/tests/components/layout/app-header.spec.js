import Component from '@/components/layout/app-header.vue'
import { event } from '@/utils'
import { mock } from '@/tests/__helpers__'

describe('components/layout/app-header', () => {
  afterEach(() => {
    jest.resetModules()
    jest.clearAllMocks()
  })

  it('renders propery', async done => {
    const wrapper = await mount(Component)
    Vue.nextTick(() => {
      expect(wrapper).toMatchSnapshot()
      done()
    })
  })

  it('toggles sidebar', () => {
    const m = mock(event, 'emit')
    shallow(Component).click('.hamburger')
    expect(m).toHaveBeenCalledWith('TOGGLE_SIDEBAR')
  })

  it('toggles search form', () => {
    const m = mock(event, 'emit')
    shallow(Component).click('.magnifier')
    expect(m).toHaveBeenCalledWith('TOGGLE_SEARCH_FORM')
  })
})
