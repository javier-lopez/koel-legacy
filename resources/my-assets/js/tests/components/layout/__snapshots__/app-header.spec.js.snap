// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/layout/app-header renders propery 1`] = `
<header id="mainHeader">
  <h1 class="brand">Koel</h1> <span class="hamburger"><i class="fa fa-bars"></i></span> <span class="magnifier"><i class="fa fa-search"></i></span>
  <div id="searchForm" class="side search showing"><input type="search" placeholder="Search" class=""></div> <span id="userBadge" class="profile"><a href="#!/profile" class="view-profile control"><img src="" alt="Avatar" class="avatar"> <span class="name"></span></a> <a class="logout"><i class="fa fa-sign-out control"></i></a></span>
</header>
`;
